DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/mmhost/src/part_1_pointers.cpp

```diff
@@ -0,0 +1,90 @@
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+using namespace sycl;
+
+struct PointerIP{
+  int *x; 
+  int *y; 
+  int *z;
+  int size;
+
+  void operator()() const {
+    for (int i = 0; i < size; ++i) {
+      z[i] = x[i] + y[i];
+    }
+  }
+};
+
+int main(void){
+#if FPGA_SIMULATOR
+  auto selector = sycl::ext::intel::fpga_simulator_selector_v;
+#elif FPGA_HARDWARE
+  auto selector = sycl::ext::intel::fpga_selector_v;
+#else  // #if FPGA_EMULATOR
+  auto selector = sycl::ext::intel::fpga_emulator_selector_v;
+#endif
+
+  bool passed = true;
+
+  try {
+    // create the device queue
+    sycl::queue q(selector, fpga_tools::exception_handler, sycl::property::queue::enable_profiling{});
+
+    // make sure the device supports USM host allocations
```

Comment:
Line 37 does not check that the device supports USM host allocations. this comment should move to line 44